Sources/Subs-Calendar.php

```diff
@@ -1025,6 +1143,302 @@ function getEventProperties($event_id)
+ * Makes sure events do not exceed the maximum allowed duration (if any).
+ * If passed an array that defines any time or date parameters, they will be used. Otherwise, gets the values from $_POST.
+ *
+ * @param array $eventOptions An array of optional time and date parameters (span, start_year, end_month, etc., etc.)
+ * @return array An array containing $start_date, $end_date, $start_time, $end_time
+ */
+function setEventStartEnd($eventOptions = array())
+{
+	global $modSettings, $user_info;
+
+	// Set $span, in case we need it
+	$span = isset($eventOptions['span']) ? $eventOptions['span'] : (isset($_POST['span']) ? $_POST['span'] : 0);
+	if ($span > 0)
+		$span = !empty($modSettings['cal_maxspan']) ? min($modSettings['cal_maxspan'], $span - 1) : $span - 1;
+
+	// Define the timezone for this event, falling back to the default if not provided
+	if (!empty($eventOptions['tz']) && in_array($eventOptions['tz'], timezone_identifiers_list()))
+		$tz = $eventOptions['tz'];
+	elseif (!empty($_POST['tz']) && in_array($_POST['tz'], timezone_identifiers_list()))
+		$tz = $_POST['tz'];
+	else
+		$tz = getUserTimezone();
+
+	// Is this supposed to be an all day event, or should it have specific start and end times?
+	if (isset($eventOptions['allday']))
+		$allday = $eventOptions['allday'];
+	elseif (empty($_POST['allday']))
+		$allday = false;
+	else
+		$allday = true;
+
+	// Input might come as individual parameters...
+	$start_year = isset($eventOptions['year']) ? $eventOptions['year'] : (isset($_POST['year']) ? $_POST['year'] : null);
+	$start_month = isset($eventOptions['month']) ? $eventOptions['month'] : (isset($_POST['month']) ? $_POST['month'] : null);
+	$start_day = isset($eventOptions['day']) ? $eventOptions['day'] : (isset($_POST['day']) ? $_POST['day'] : null);
+	$start_hour = isset($eventOptions['hour']) ? $eventOptions['hour'] : (isset($_POST['hour']) ? $_POST['hour'] : null);
+	$start_minute = isset($eventOptions['minute']) ? $eventOptions['minute'] : (isset($_POST['minute']) ? $_POST['minute'] : null);
+	$start_second = isset($eventOptions['second']) ? $eventOptions['second'] : (isset($_POST['second']) ? $_POST['second'] : null);
+	$end_year = isset($eventOptions['end_year']) ? $eventOptions['end_year'] : (isset($_POST['end_year']) ? $_POST['end_year'] : null);
+	$end_month = isset($eventOptions['end_month']) ? $eventOptions['end_month'] : (isset($_POST['end_month']) ? $_POST['end_month'] : null);
+	$end_day = isset($eventOptions['end_day']) ? $eventOptions['end_day'] : (isset($_POST['end_day']) ? $_POST['end_day'] : null);
+	$end_hour = isset($eventOptions['end_hour']) ? $eventOptions['end_hour'] : (isset($_POST['end_hour']) ? $_POST['end_hour'] : null);
+	$end_minute = isset($eventOptions['end_minute']) ? $eventOptions['end_minute'] : (isset($_POST['end_minute']) ? $_POST['end_minute'] : null);
+	$end_second = isset($eventOptions['end_second']) ? $eventOptions['end_second'] : (isset($_POST['end_second']) ? $_POST['end_second'] : null);
```

Comment:
Yea, I was thinking more along the lines of assuring they are ints in most causes (as day, hour, minute and second should always be this) and other simple validation.  I'm not sure what dirty input into strtotime would cause, but it seems we should only make sure sensible data should even make it there.  Since you already setting up the vars, this isn't too much other than (int) on most everything.